### PR TITLE
Changes for orcacode experiment

### PR DESCRIPTION
**Pull request by andreaskoepf**

### CODE REVIEW

**Comment by shahules786** [comment]
Besides this I think there is a bug in this [line](https://github.com/LAION-AI/Open-Assistant/blob/9c4eafca993f8effcd08f59d08864552dcc6999b/model/model_training/models/patching.py#L250). Config cannot be updated. I removed it from my version. Can you check that too with this PR? @andreaskoepf

File: model/model_training/custom_datasets/prompt_dialogue.py [hunk 1]
```diff
@@ -199,45 +198,37 @@ def __getitem__(self, idx):
 class DolphinMix(Dataset):
     name = "dophin-mix"
 
-    def __init__(self, cache_dir, num_samples=100000, max_char_len=8000, seed=42):
+    def __init__(self, cache_dir, num_samples: Optional[int] = None, max_char_len: int = 8000, seed: int = 42):
```
Comment (shahules786) [inline model/model_training/custom_datasets/prompt_dialogue.py:201]: Isn't it better to add `data_files` as an argument with the default value "flan-5m..."?  In this way, we can change to gpt-4 version using config if required. 
`

File: model/model_training/models/patching.py
Comment (shahules786) [inline model/model_training/models/patching.py:1]: Can you remove the `update_config` function definition too since it's not used?

**Review by shahules786 — CHANGES_REQUESTED** [review]

**Comment by shahules786** [comment]
Nice, thanks.

**Review by shahules786 — APPROVED** [review]

**Review by s0mt0chukwu — APPROVED** [review]

**Pull request merged**